src/ima-parser.js

```diff
@@ -7,6 +9,41 @@ export default class IMAParserPlugin extends CorePlugin {
 
   constructor(core) {
     super(core)
-    console.log(`${this.name} plugin loaded!`)
+    this._VMAPHandler = new VMAPManager()
+    this._VASTHandler = new VASTManager()
+  }
+
+  /**
+   * Get a list of AdBreaks from the received URL.
+   * @param {String} url Ad Server URL to fetch VMAP XML.
+   * @returns {Promise} Promise resolved with one array of AdBreaks or one error.
+   */
+  requestAdBreaks(url) {
+    return this._VMAPHandler.request(url)
```

Comment:
Thinking as a client of this class, wouldn't be better to just call `VMAPManager.request(url)` and get all the ad breaks parsed without having to know that we should also chain calls to `process` and `filterRawData`? These last two would be better off as private methods of the class.

Reply:
It's just that both the `VMAPManager` and the `VASTManager` won't be public (they won't be exported in the final build and their internal references are private `_*`).

Since the only `IMAParserPlugin` will be dealt with, I had no problem making the process of getting `AdBreak` more granular.